pyccel/parser/semantic.py

```diff
@@ -357,6 +368,32 @@ def check_for_variable(self, name):
         else:
             return self.scope.find(name, 'variables')
 
+    def check_for_attribute(self, name):
+        """
+        Search for an Attribute with the given name in the current scope.
+
+        Search for an Attribute with the given name in the current scope,
+        defined by the local and global Python scopes. Return None if not found.
+
+        Parameters
+        ----------
+        name : DottedName
+            The object describing the attribute.
+
+        Returns
+        -------
+        Variable
+            Returns the attribute if found or None.
+        """
+
+        prefix = self._visit(DottedName(*name.name[:-1]))
```

Comment:
Does the class `DottedName` return a PyccelSymbol if you only pass one element?

Reply:
Fixed.

Reply:
When possible ([as mentioned in the docs](https://github.com/pyccel/pyccel/blob/devel/developer_docs/review_process.md?plain=1#L25C116-L25C192)), please leave references to commits or code when you say something is fixed so it is easy to check.